Update PULL_REQUEST_TEMPLATE.md (#17669)
Azure/azure-rest-api-specs: .github/PULL_REQUEST_TEMPLATE.md

## ARM API Information (Control Plane)

> <details>

Reviewer: Do we really want to these to be collapsed sections? I worry a little that folks will not ever expand them and read or check any of the options.

Author: Yeah, I see that point. @markweitzel any thoughts?

Author: The collapsible sections do have the advantage of focusing people's attention on ARM vs Data plane until we are able to have 2 templates.

      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist (MS Employees Only):

> - [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"

Reviewer 2: The "bcforapi" short link points to an outdated version of the breaking changes doc. I suggest using the link that Jeffrey/Srikanth maintain to always point to the latest version.
```suggestion
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
```

> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:

> - [ ] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.

Reviewer 3: Hi @czubair , can you revert the change of quotation mark coz it will break the automation tool?

> - [ ] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.

Author: ```suggestion
- [ ] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
```

  - Adding a new service
  - Adding new API(s)
  - Adding a new API version

>     -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

Reviewer 4: Besides adding this detail guide, Can we add back Openapi hub link here which can help service team easy to create this kind of PR?

>     -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

Author: ```suggestion
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).
```

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If you have any breaking changes as defined in the [Breaking Change Policy](http://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.
  
**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).
  

> NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Reviewer 5: Is this message important (as the link seems also private)?

Reviewer 2: I think this policy is not widely / well understood so we should keep it.

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

## Non-ARM API Information (Data Plane)

This PR template is for Data Plane APIs only.

### API Info: The Basics

* Link to engagement record issue (if known): 

Is this review for (select one):

> - [ ] a public preview

Reviewer 5: ```suggestion
- [ ] A public preview
```

Reviewer 5: Could there be a private preview? I assume this template will be shared in private repo as well, and team may use that repo for private preview.

- [ ] GA release 

> 

Reviewer 5: Should we also add a check list like below?

```
### REST API Review Checklist 

- [ ] Ensure you've reviewed [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). This is required before you can request review from Azure REST API Stewardship.
```

I see it is mentioned once in `ARM API Information` section and also included indirectivity in contact section. But we might want to emphasis the Guidelines again.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If you have any breaking changes as defined in the [Breaking Change Policy](http://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.
  
**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).
  
NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).
  
</details>

> ## non-ARM/ Data Plane API

Author: ```suggestion
## Non-ARM API Information (Data Plane)
```

<details>
  <summary>Expand this section for dataplane</summary>

This PR template is for Data Plane APIs only.

### API Info: The Basics

> * Link to engagement record issue: 

Author: ```suggestion
* Link to engagement record issue (if known): 
```

- [ ] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 

> If you have any breaking changes as defined in the [Breaking Change Policy](http://aka.ms/bcforapi), request approval from the Breaking Change Review Board.

Reviewer 2: Here too.
```suggestion
If you have any breaking changes as defined in the [Breaking Change Policy](http://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.
```

  
**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).
  
NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).
  
</details>

## Data Plane API

<details>
  <summary>Expand this section for dataplane</summary>

This PR template is for Data Plane APIs only.

### API Info: The Basics

> Most of the information about your service should be captured in the issue that serves as your [*engagement record*](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/271/Azure-REST-API-Stewardship?anchor=rest-api-stewardship-process).

Author: ```suggestion

```

* Design Document: 
* Previous Open API Doc: 
* Updated paths:
  
### :grey_question: Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azapi) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azapi).
* :speech_balloon: [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* :love_letter: [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>
  
### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/azapi/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications
 * [Azure REST API Guidelines](https://aka.ms/azapi/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/azapi/style)

>  * [Azure Breaking Change Policy](http://aka.ms/bcforapi)

Reviewer 2: Here too.
```suggestion
 * [Azure Breaking Change Policy](http://aka.ms/AzBreakingChangesPolicy/)
```

  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If you have any breaking changes as defined in the [Breaking Change Policy](http://aka.ms/bcforapi), request approval from the Breaking Change Review Board.
  

> **Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Reviewer: Do we really want to link to internal only documentation here?

Author: @JeffreyRichter are you okay with using these links or do you find it problematic?

Reviewer 6: I'm fine with it as long as external people can't access these docs.
We need to cater to our internal teams as best as possible.

Author: @weshaggard it seems these links would work as we need them to - to reach internal folks only.

Reviewer: @JeffreyRichter is there any reason these cannot or shouldn't be in our public specs repo?

Reviewer 6: Yes, the http://aka.ms/bcforapi PDF file has employee names in it related to legal and marketing and who signs off on breaking changes.

The wiki has internal comments/discussion about proposed changes to policies which must not be external.

Reviewer: OK so there might be some things we need to protect but for an open source project there are a lot of links that are going to 404 in this common PR template which is not a great experience.  As a potential future improvement, it would be good to separate these into public and private and keep the private links to a minimum (ideally no more than 1).

Reviewer 7: Hi Wes, I agree with you.  We can prioritize the efforts when we working on to support 3rd party resource provider.

  
NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).
  
</details>

## Data Plane API

<details>
  <summary>Expand this section for dataplane</summary>

This PR template is for Data Plane APIs only.

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*engagement record*](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/271/Azure-REST-API-Stewardship?anchor=rest-api-stewardship-process).

* Link to engagement record issue: 

Is this review for (select one):

> - [ ] a private preview

Reviewer 2: Should we even list "private preview" as an option for the public repo?

> - [ ] a private preview

Author: ```suggestion
```

- [ ] a public preview
- [ ] GA release 

### What are you changing?
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous Open API document (swagger) if applicable, and the root paths that have been updated. </sup>

#### Description
< Your text here >
  
#### References & Links  
* Design Document: 
* Previous Open API Doc: 
* Updated paths:
  

> ## :grey_question: Got questions? Need additional info?? We are here to help!

Reviewer 2: Should be subheading of "Data Plane API"
```suggestion
### :grey_question: Got questions? Need additional info?? We are here to help!
```

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azapi) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azapi).
* :speech_balloon: [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* :love_letter: [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>
  
### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)

>  * [Spectral Linting](https://aka.ms/style)

Reviewer: I don't think this is the correct link, also why are we linking spectral? Aren't we still using OAV in our pipelines?

Reviewer 2: OAV is used in our pipelines but we use Spectral in the API reviews (in addition to all the checks done in the pipeline) so we want teams to be aware of it.

Reviewer 2: ```suggestion
 * [Spectral Linting](https://aka.ms/azapi/style)
```

 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications

>  * [Azure REST API Guidelines](https://aka.ms/guidelines)

Reviewer: @czubair have you validated all these links? I haven't check them all but at least this one doesn't work.

>  * [Azure REST API Guidelines](https://aka.ms/guidelines)

Reviewer 2: ```suggestion
 * [Azure REST API Guidelines](https://aka.ms/azapi/guidelines)
```

>  * [OpenAPI Style Guidelines](https://aka.ms/style)

Reviewer 2: ```suggestion
 * [OpenAPI Style Guidelines](https://aka.ms/azapi/style)
```

  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

> - [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

Author: ```suggestion
- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc). This is required before you can request review from ARM API Review board.
```
